deltacat/storage/README.md

```diff
@@ -0,0 +1,100 @@
+
+The canonical string is composed of its parent's canonical string, plus its own canonical string elements (found in the LocatorName `parts`).
+
+| **Object Type**    | **Primary Key**                          | **Mutability**  |
+|---------------------|------------------------------------------|-----------------|
+| **Namespace**       | `name`                                  | Mutable         |
+| **Table**           | `name`                                  | Mutable         |
+| **Table Version**   | `name`                                  | Immutable       |
+| **Stream**          | `stream_id` + `format`                  | Immutable       |
+| **Partition**       | `partition_id` + `partition_values`     | Immutable       |
+| **Delta**           | `stream_position`                       | Immutable    
+
+# Directory/file structure
+## Object Directories (Immutable id)
+Every metastore object (Namespace, Table, TableVersion, etc.) has a root 
+directory. 
+
+named by its **immutable ID** (e.g., a UUID or a numeric ID). 
+Inside that directory:
+
+For instance, a namespace root looks like:
+```
+${catalog_root}/${namespace_id}/
+    └──rev/
+        ...
+    └──${id-table-1}/
+        ...
+    └──${id-table-2}/
+        ...
+```
+**Child object directories**
+
+The parent object has a single directory for each child object directory. Those directories in turn use the structure described above.
+
+**Revision Directory**
+The **`rev/`** directory contains versioned metadata files named:
+`<revision_number_padded_20_digits>_<txn_operation_type>_<txn_id>.mpk`
+
+- `revision_number_padded_20_digits` is zero-padded, e.g., `00000000000000000001`.
+- `txn_operation_type` is typically `create`, `update`, or `delete`.
+- `txn_id` is a unique identifier for the transaction that performed this operation.
+
+### Mutable Name Directories
+Certain objects (like **Namespaces** and **Tables**) may have **mutable** names. 
+To support renames while keeping an **immutable** ID, the code can create
+a “digest-based” directory:
+
+- The **digest** is a SHA-1 hash of the object’s **canonical locator** string 
+  (e.g., `"MyNamespace"` or `"MyNamespace|MyTable"`).
+- This directory name is simply that SHA-1 digest (hex-encoded).
+- Inside this **mutable name directory**, there is a single **Name Mapping File** 
+  (zero bytes) that references the **immutable ID** directory.
+
+**Name Mapping File**
+
+The mutable name directory just contains a file which maps to its immutable directory. The format of this file is: <revision_number_padded_20_digits>_<txn_operation_type>_<txn_id>.<object_id>
+
+### Transaction Directory
+The transaction directory (/txn) is a special directory in the catalog root which holds all successfully committed transactions. It contains one empty file per successful transaction.
```

Comment:
Transaction files will be populated in my next PR, so may as well update this now:

```suggestion
The transaction directory (${catalog_root}/txn) is a special directory in the catalog root which holds all successfully committed transactions. It contains one file per successful transaction, whose contents record details about all operations fulfilled by the transaction.
```